From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Wed, 17 Feb 2021 15:11:18 -0500
Subject: Simplfy use of pkg-config when testing for EFL

I do not know whether this is the right approach, but
we are not building the enlightenment pinentry anyway, yet.

Forwarded: https://dev.gnupg.org/T5308
---
 configure.ac | 26 +-------------------------
 1 file changed, 1 insertion(+), 25 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7315f2c..0833e4f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -420,32 +420,8 @@ AC_ARG_ENABLE(pinentry-efl,
             AS_HELP_STRING([--enable-pinentry-efl],[build EFL pinentry]),
             pinentry_efl=$enableval, pinentry_efl=maybe)
 
-dnl check for pkg-config
 if test "$pinentry_efl" != "no"; then
-	AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
-	if test x"${PKG_CONFIG}" = xno ; then
-		pinentry_efl=no
-	fi
-fi
-
-if test "$pinentry_efl" != "no"; then
-	AC_MSG_CHECKING([for efl])
-	"${PKG_CONFIG}" --exists 'elementary >= 1.18'
-	if test $? -ne 0 ; then
-		AC_MSG_RESULT([no])
-		AC_MSG_WARN([efl >= 1.18 is required for efl pinentry])
-		pinentry_efl=no
-	else
-		AC_MSG_RESULT([yes])
-		EFL_CFLAGS=`"${PKG_CONFIG}" --cflags ecore-x elementary`
-		EFL_LIBS=`"${PKG_CONFIG}" --libs ecore-x elementary`
-		AC_SUBST(EFL_CFLAGS)
-		AC_SUBST(EFL_LIBS)
-		if test "$pinentry_efl" != "no"
-		then
-			pinentry_efl=yes
-		fi
-	fi
+	PKG_CHECK_MODULES(efl,[elementary > 1.18])
 fi
 AM_CONDITIONAL(BUILD_PINENTRY_EFL, test "$pinentry_efl" = "yes")
 
